isotovideo

```diff
@@ -323,6 +326,29 @@ our $tags = undef;
 my $backend_requester = undef;
 
 my ($last_check_seconds, $last_check_microseconds) = gettimeofday;
+
+# Kill all the descendantes of a PID - A grandchild process is not considered a child
+# We loop through the proc instead of checking the task/tid/children, so we can invoke this when processing SIGCHLD.
+# Since the orphans are adopted by init (pid 1), we need to check if the process group is the same as the actual group (since forks have the same group)
+# To know more about procfs, please check man(5) proc
+sub _kill_all_descendants {
+    my ($ppid) = @_;
+    my $group = getpgrp(0);
+    for my $proc_folder (glob("/proc/*")) {
+        next if (!-d $proc_folder || !-f "$proc_folder/stat" || $proc_folder eq "/proc/$$");
+        open my $ifh, '<', "$proc_folder/stat";
```

Comment:
What would happen here if the filehandle fails to be opened?

Reply:
@mudler 

you are interested only in the processes you own, and in those you always have permissions to read, but the the stat file is always available to be read (the /proc/[pid]/status one is protected).

But you have a point. Let's say that a process dies, between the `next`and the `open` :-)

Let me fix it :-)

Reply:
Whatever you are reading, Perl can fail opening a filehandle

Reply:
@mudler done